src/clj/ataru/virkailija/virkailija_routes.clj

```diff
@@ -98,8 +99,10 @@
                  (api/POST "/forms" {session :session}
                    :summary "Persist changed form."
                    :body [form ataru-schema/FormWithContent]
-                   (trying #(form-store/upsert-form
-                             (assoc form :modified-by (-> session :identity :username)))))
+                   (match (trying #(form-store/create-form-or-increment-version!
```

Comment:
its wrapped like this because I wanted to keep the regular response handling (what trying does) and in case there was a concurrently-modified conflict rewrite the response as a bad-request instead.

Before I tried throwing an exception but it isn't exactly exceptional when somebody got to the db first.